scanner/src/test/kotlin/utils/FileListResolverTest.kt

```diff
@@ -57,7 +57,20 @@ class FileListResolverTest : StringSpec({
 
         val fileList = resolver.resolve(ArtifactProvenance(sourceArtifact = RemoteArtifact.EMPTY))
 
-        fileList.toYaml() shouldBe File("src/test/assets/expected-file-list.yml").readText()
+        fileList.ignorePatterns should containExactlyInAnyOrder(
```

Comment:
Why is independence of serialization valuable here?

Reply:
It makes the test file self-contained and does not require IO for the verification.

Reply:
> It makes the test file self-contained and does not require IO for the verification.
hm,
> Why is independence of serialization valuable here? 
...compared to the other cases where we use expected results.

Reply:
Yes, but these are often files with hundreds of lines (like for the package managers tests) where inlining would make the test class unreadable. In this case I find it more convenient not to have to look at a separate file.
If you think we should keep testing serialization, especially if the ordering works correctly, I would rather add a separate test case for that (but probably also not use an external file).

Reply:
See #9540.

Reply:
I've found the separate file to be easier to read, but anyhow - no agreement needed on this.